lib/denv/sources/rails_credentials.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+class DEnv
+  class Sources
+    # Class for loading credentials specifically from Rails
+    class RailsCredentials < Base
+      attr_reader :credentials_hash, :credentials_location
+
+      def initialize(credentials_hash, credentials_location)
+        @credentials_hash = credentials_hash || {}
+        @credentials_location = credentials_location
+      end
+
+      def type
+        'credentials'
+      end
+
+      def key
+        "#{type[0, 1]}:credentials"
+      end
+
+      def reload
+        load_new_creds
+        super
+      end
+
+      private
+
+      def load_new_creds
+        return nil if credentials_location.nil? || credentials_location == ''
+        return nil unless Object.const_defined?(:Rails)
```

Comment:
@aemadrid here is a check for if rails is defined or not.

Reply:
shouldn't we throw an exception instead of silently failing?

Reply:
I like that idea more. Making that change

Reply:
@aemadrid what are your thoughts on only allowing this feature if they are in a rails app? or should we still allow it but prevent the reloading?